src/plugins/discord/fetcher.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+
+import * as Model from "./models";
+import * as NullUtil from "../../util/null";
+
+type FetchEndpoint = (endpoint: string) => Promise<any>;
+
+/**
+ * Provide the Guild ID to fetch against, and the 'limit'
+ * parameter when fetching GuildMembers, Messages, and Reactions.
+ *
+ */
+type FetchOptions = {|
+  guildId: Model.Snowflake,
+  membersLimit: number,
+  messagesLimit: number,
+  reactionsLimit: number,
+|};
+
+export type EndCursor = Model.Snowflake | null;
```

Comment:
The way EndCursor is handled seems inconsistently. We declare that the type is either Snowflake or null, but the methods which take `EndCursor`s error on the null case. 

Therefore I propose either:

- We make the EndCursor type just a snowflake, and update the PageInfo so that it has `endCursor | null`
or:
- We make the methods that take EndCursor cleanly accept the null case (by just not appending the `after` flag)

Reply:
Updated, good suggestion